### PR TITLE
GH-320 Reorder potential library paths for macOS Monterey compatibility.

### PR DESCRIPTION
**Pull request by h-vn**
Running 'perl Makefile.PL' may fail with:
   WARNING: .../perl is loading libcrypto in an unsafe way

See https://github.com/radiator-software/p5-net-ssleay/issues/329
and https://openradar.appspot.com/FB9725981 (linked by GH-329) for
more information.

To summarise why this commit helps in this case:

  The failure is triggered when the prospective library directory exists but
  does not contain the desired library file. That is, $prefix/lib64 does not
  trigger the failure because the directory pointed by OPENSSL_PREFIX does not
  contain lib64 subdirectory.

### CODE REVIEW

File: Makefile.PL [hunk 1]
```diff
@@ -206,7 +206,8 @@ sub ssleay_get_build_opts {
         }
     }
 
-    for ($prefix, "$prefix/lib64", "$prefix/lib", "$prefix/out32dll") {
+    # Note: the order matters for macOS Monterey.
```
Comment (chrisnovakovic) [inline Makefile.PL:209]:
```suggestion
    # Note: the order matters for macOS Monterey. See GH-329.
```

Might be worth referencing #329 here, in case someone wants to change the order again in future and wonders why Monterey is an edge case.

**Review by chrisnovakovic — APPROVED** [review]
Since our CI setup doesn't really cover this, I don't think there's much more we can do to test the effect this will have on end users than to do a new developer release and see if we get back any smoke test failures...

File: Makefile.PL (hunk 1 above)
Comment (h-vn) [reply Makefile.PL:209]: Appveyor shows that it now fails to link with recent Strawberry Perls. I'll work on this more and also add pointers to further information. I'll also add a `Changes` entry.

**Comment by h-vn** [comment]
The diff looks ugly. Redo with the same content using a different pull request
